report/4.Solution/SemanticsDescription.tex

```diff
@@ -0,0 +1,748 @@
+\begin{center}
+	\begin{math}
+	\cfrac
+	{env_v', env_P'[f \mapsto (L, e, env_V', env_P')] \vdash \langle L, sto[l \mapsto v] \rangle \rightarrow_{stm} v}
+	{env_v, sto \vdash f \rightarrow_{F_{exp}} v}
+	\end{math}
+	
+	\texttt{ where } $env_P\ f = (L, e, env_V', env_P')$\\
+	\texttt{ and } $ env_V', sto \vdash e \rightarrow_{e} v$\\
+\end{center}
+
+\begin{center}
+	\begin{math}
+	\cfrac
+	{env_v'[x \mapsto l][nest \mapsto new\ l], env_P'[f \mapsto (L, x\sim, e, env_V', env_P')] \vdash \langle L, sto[l \mapsto v] \rangle \rightarrow_{stm} sto'}
+	{env_v, sto \vdash x f x\sim \rightarrow_{F_{exp}} v}
+	\end{math}
+	
+	\texttt{ where } $env_P\ f = (L, e, env_V', env_P')$\\
+	\texttt{ and } $ env_V', sto \vdash e \rightarrow_{e} v$\\
+	\texttt{ and } $l = env_V\ next$
+\end{center}
+
+\subsection{Big-Step Semantic for Variable Declarations}
+Variable declarations $Dcl_V$ can change the variable environment $Env_v$ by declaring new variables in $Var$.
+They can also change the store by binding variables to new addresses.
+
+The transition-system for $Dcl_V$ is ($(Dcl_V\ X\ Env_V\ X\ Sto) \cup Env_V\ X\ Sto, \rightarrow_{D_{V}}, Env_V\ X\ Sto$).
+
+This means that the transitions must be on the form.
+
+\begin{center}
+	$\langle D_V,env_V,sto \rangle \rightarrow_{D_V} (env_V^{'} , sto^{'})$
+\end{center}
+
+The transitions are shown below.
+
+\textbf{[$Dcl_VAssExp_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		{\langle T\ x = e;\ S,\ env_V,\ sto \rangle \rightarrow_{D_V} (env_V'[x \mapsto l][next \mapsto l'],\ sto')}
+	\end{math}
+	
+	\texttt{ where } $env_V,\ sto \vdash e \rightarrow_e v$
+	\texttt{ and } $l = env_V\ next$
+	\texttt{ and } $l' = new\ l$
+\end{center}
+
+\texttt{[$Dcl_VAssFnc_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		{\langle T\ x = f,\ env_V,\ sto \rangle \rightarrow_{D_V} (env_V'[x \mapsto l][next \mapsto l'],\ sto')}		
+	\end{math}
+	
+	\texttt{ where } $env_V,\ sto \vdash f \rightarrow_{call_{exp}} v$
+	\texttt{ and } $l = env_V\ next$
+	\texttt{ and } $l' = new\ l$
+\end{center}
+
+\textbf{[$Dcl_VNoAss_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		{\langle T\ x;\ S,\ env_v,\ sto \rangle \rightarrow_{D_V} (env_v'[x \mapsto l][next \mapsto l'],\ sto)}
+	\end{math}
+	
+	\texttt{ where } $l = env_v\ next$
+	\texttt{ and } $l' = new\ l$
+\end{center}
+
+\textbf{[$Dcl_VEmpty_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	\langle \epsilon,\ env_v,\ sto \rangle \rightarrow (env_v,\ sto)
+	\end{math}
+\end{center}
+
+\subsection{Big-Step Semantics for Procedure and Function Declarations}
+Function declarations $Dcl_P$ can change the procedure environment $Env_P$ by declaring new functions in $Fnames$.
+
+The transition-system for $Dcl_P$ is ($Fnames \rightharpoonup Lambda\ X\ Var\ X\ Exp\ X\ Env_V\ X\ Env_P$).
+This means that the transitions must be on the form.
+
+\begin{center}
+	$\langle D_P, env_V, env_P \rangle \rightarrow_{D_P} (env_P)$
+\end{center}
+
+Procedures are functions with no return value and therefore are statements.
```

Comment:
maybe explain this before you explain procedures and functions at all

Reply:
Fixed